### DISCUSSION TITLE
Some ideas for useful enhancement

### DISCUSSION THREAD

**Discussion by Twix53791** [ahrm/sioyek#197] [Ideas]
Hello,

I just discovered Sioyek and I am gonna give it a try to read and highlights my pdf files.  I created my own config and I saw straightaway some limitations which could be enhanced, I think

1) Visual signs/way to visualize the marks, bookmarks and portals on the document : it seems to me a bit weird to set invisible things on a document ! How to remember where I have put marks or portals? I didn't find an easy way to display that. It could be a mode to toggle, or a permanent small colored mark. 

2) Put comments on the documents : bookmarks maybe are supposed to emulate such a feature? I think a way of adding comments easily displayed (for example via the overview?) could be much better. The bookmark just allow us to jump to a location specified by a text, but a comment is a (perhaps long) text we put somewhere to note something, which have to be after easily displayed, to be read along the document text.

3) A shortcut to close the helper window quickly? I didn't find a way to do it via sioyek shorcuts. q doesn't works...

4) A way to open sioyek with some "modes" (like toggle_horizontal_scroll_lock or toggle_dark_mode) by default? For example, I would like to have the dark mode by default personally. A bit a bore to have to press a key to toggle it when I open sioyek.

**Comment by xulongwu4** [comment]
I believe your last point is already addressed by the `startup-commands` option in the configuration: https://sioyek-documentation.readthedocs.io/en/latest/configuration.html#startup-commands

**Reply by Twix53791** [reply]
good point

**Comment by ahrm** [comment]
> A shortcut to close the helper window quickly? I didn't find a way to do it via sioyek shorcuts. q doesn't works...

You can do that with `toggle_one_window` command (`f12` by default)

>A way to open sioyek with some "modes" (like toggle_horizontal_scroll_lock or toggle_dark_mode) by default? For example, I would like to have the dark mode by default personally. A bit a bore to have to press a key to toggle it when I open sioyek.

As @xulongwu4 mentioned, you can do that with `startup_commands`.
